### PR TITLE
Preprocessing of train/test images + Two localizer networks

### PR DESCRIPTION
**Pull request by opraveen**
Preprocessing tested locally on CPU. Haven't tested or run the new localizer networks on GPU. The two solvers in run.sh are still commented out in this version

### CODE REVIEW

File: train/point1_train.prototxt [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+  top: "conv8"
+  convolution_param {
+    num_output: 16
+    kernel_size: 3
+    weight_filler {
+      type: "gaussian"
+      std: 0.1
+    }
+  }
+}
```
Comment (sjuvekar) [inline train/point1_train.prototxt:226]: I missed Adding a BatchNorm layer "conv8_bn" here. Can you please add it?

**Comment by sjuvekar** [comment]
I get this Caffe error after adding the missing BatchNorm layer: 

layer.hpp:389] Check failed: ExactNumTopBlobs() == top.size() (2 vs. 1) ImageData Layer produces 2 top blob(s) as output.

File: train/point1_train.prototxt (hunk 1 above)
Comment (opraveen) [inline train/point1_train.prototxt:226]: I can add this, I am also trying to fix the other errors. Not sure, but I think we may have to convert images to leveldb/lmdb format (using create_imagenet.sh*)
- http://caffe.berkeleyvision.org/gathered/examples/imagenet.html

**Comment by opraveen** [comment]
I made some updates for localizer networks in here: https://github.com/opraveen/Kaggle-Whales-Caffe/commit/6a0e70c6f28002ed3f0f886a04c33d1d7ca65d3f

However, I am yet to test these on a gpu. This now uses lmdb format and the errors are gone while running on a cpu

**Comment by sjuvekar** [comment]
I am paying for g2.2xlarge myself. I created a fresh instance and installed
all tools on my own.

I'll checkout the commit and try running it sometime today

Thanks,
Sudeep

**Pull request merged**